client/src/services/ServiceWorker/Controllers/PopupController.js

```diff
@@ -0,0 +1,257 @@
+        this.openPopup({ url: origin, type: POPUP_TYPE.CONNECT_ACCOUNT });
+    }
+
+    openSignRequest = async ({ origin }) => {
+        this.openPopup({ url: origin, type: POPUP_TYPE.SIGN_DEPLOY });
+    }
+
+    openSignMessageRequest = async ({ origin }) => {
+        this.openPopup({ url: origin, type: POPUP_TYPE.SIGN_MESSAGE });
+    }
+
+    openPopup = async ({ url, type }) => {
+        if (this.popupWindow) {
+            try {
+                let window = await browser.windows.get(this.popupWindow.windowId);
+                if (window.id) {
+                    browser.windows.update(window.id, {
+                        focused: true,
+                        drawAttention: true
+                    });
+
+                    return false;
+                }
+                return this.popupWindow.windowId;
+            } catch(e) {
+                // eslint-disable-next-line no-console
+                console.log(e);
+            }
+        }
+
+        const window = await browser.windows.getCurrent();
+        const xOffset = window.left ?? 0;
+        const yOffset = window.top ?? 0;
+
+        const createdWindow = await browser.windows.create({
+            url: `/popup.html#/${type}`,
+            type: 'popup',
+            height: 480,
+            width: 300,
+            left: xOffset,
+            top: yOffset
+        });
+
+        this.currentSite = url;
+        this.popupWindow = {
+            windowId: createdWindow.id,
+        };
+
+        return url;
+    }
+
+    closePopup = async ({windowId} = {}) => {
+        if (windowId) {
+            await browser.windows.remove(windowId);
+        } else {
+            let currentWindow = await browser.windows.getCurrent();
+            if (currentWindow.type === 'popup' && currentWindow.id) {
+                await browser.windows.remove(currentWindow.id);
+            }
+        }
+
+        this.popupWindow = null;
+    }
+
+    isConnected = async ({ origin } = {}) => {
+        const currentPublicKey = await this.accountController.getCurrentPublicKey();
+        if (!currentPublicKey) {
+            return false;
+        }
+
+        if (!this.currentSite && origin) {
+            this.currentSite = origin;
+        }
+
+        const connectedSites = await this.getConnectedSites();
+
+        if (_isEmpty(_get(connectedSites, currentPublicKey, false))) {
+            return false;
+        }
+
+        return !!connectedSites[currentPublicKey].find(connectedSite => connectedSite === this.currentSite);
+    }
+
+    getCurrentSite = () => {
+        return this.currentSite;
+    }
+
+    getConnectedSites = async () => {
+        const valueObj = await getChromeStorageLocal(CONNECTED_SITES);
+
+        return _get(valueObj, CONNECTED_SITES, {});
+    }
+
+    addConnectedSite = async ({ site, publicKey }) => {
+        let connectedSites = await this.getConnectedSites();
+        const sites = _get(connectedSites, publicKey, false);
```

Comment:
Minor: Should not mix variable type. I'm assuming the default value here would be `undefined` or empty array `[ ]`.